File: throwback-webui/src/app/objects/ContentResponse.ts
import {environment} from "../../environments/environment";

export class ContentResponse{
    title:string = "";
    creator: string = "";
    filename: string = "";
    createdDate:Date|undefined = new Date();
    description:string = "";
    url_safe_name:string = "";
    userMeta:ContentMeta



    constructor(title: string, creator: string, filename: string, createdDate: Date, description: string,url_safe_name:string) {
        this.title = title;
        this.creator = creator;
        this.filename = filename;
        this.createdDate = createdDate;
        this.description = description;
        this.url_safe_name = url_safe_name;
    }

}

export class ContentMeta{
    like_count:number
    dislike_count:number
    view_count:number
    viewer_status:number


    constructor(like_count: number, dislike_count: number, view_count: number, viewer_status: number) {
        this.like_count = like_count;
        this.dislike_count = dislike_count;
        this.view_count = view_count;
        this.viewer_status = viewer_status;
    }
}

export function makeEmptyContentResponse():ContentResponse{
    return new ContentResponse("","", "", new Date(),"","");
}

export function makeThumbnailUrl(res:ContentResponse):string{
    return environment["S3-URL"] + "/" + res.filename
}

File: throwback-api/src/content/content_response.py
import string
import datetime


class ContentUserMetaData:
    def __init__(self,like_count,dislike_count,view_count,viewer_status):
        self.like_count = like_count
        self.dislike_count =dislike_count
        self.view_count = view_count
        self.viewer_status = viewer_status

class ContentResponse:
    userMeta = {}

    def __init__(self, title:string, creator:string, s3_filename:string,
                 created_date:string, desc:string,url_safe_name:string,content_id:string):
        self.title = title
        self.filename = s3_filename
        self.creator = creator
        self.description = desc
        self.createdDate = created_date
        self.url_safe_name = url_safe_name
        self.content_id = content_id


def content_response_serialize(obj):
    if isinstance(obj, datetime.date):
        return dict(year=obj.year, month=obj.month, day=obj.day)

    return obj.__dict__


